app/dialog/dialog.js

```diff
@@ -20,23 +34,109 @@ export default class Dialog extends crsbinding.classes.ViewBase {
         const instance = this._element.querySelector("#dialog-content").content.cloneNode(true);
         const position = document.querySelector("#positionOptions").value;
 
-        const anchor = {
-            left: "top",
-            right: "top",
-            bottom: "left",
-        }
-
         await crs.call("dialog", "show", {
             title: "My Title",
             main: instance,
             target: event.target,
             position: position,
-            anchor: anchor[position],
+            anchor: this.#anchor[position],
             margin: 10,
             parent: "main"
         });
     }
 
+    async showParentDialog(event) {
+        const instance = this._element.querySelector("#nested-dialog").content.cloneNode(true);
+        const position = document.querySelector("#positionOption").value;
+
+        const header = await crs.call("dom", "create_element", {
+            tag_name: "template",
+            children: [
+                {
+                    tag_name: "h2",
+                    text_content: "&{dialog.title}"
+                },
+                {
+                    tag_name: "button",
+                    text_content: "open-fullscreen",
+                    classes: ["icon", "transparent"],
+                    attributes: {
+                        id: "btnResize",
+                        "data-action": "resize",
+                        "aria-label": "&{en.resize}"
+                    }
+                },
+                {
+                    tag_name: "button",
+                    text_content: "close",
+                    classes: ["icon", "transparent"],
+                    attributes: {
+                        id: "btnClose",
+                        "data-action": "close",
+                        "aria-label": "&{en.close}"
+                    }
+                }
+            ]
+        });
+        await crs.call("dialog", "show", {
+            main: instance,
+            title: "My Parent Dialog",
+            target: event.target,
+            position: position,
+            anchor: this.#anchor[position],
+            margin: 10,
+            parent: "main",
+            header: header.content,
+            callback: async (args) => {
+                if (args.action === "openDialog") {
+                    await this.#showChildDialog();
+                }
+            }
+        });
+    }
+
+    async #showChildDialog() {
+        const instance = this._element.querySelector("#child-dialog").content.cloneNode(true);
+
+        const header = await crs.call("dom", "create_element", {
```

Comment:
This looks like a copy-paste of the above header you're creating for the parent dialog. In this situation, please add a template on the html with a relevant id, and just retrieve the template and clone it's content for both here and the parent dialog - similar to what we're doing on line 99 above

Reply:
created the template and replaced the duplicate code.